cluster/cluster.go

```diff
@@ -210,6 +219,46 @@ func Create(
 		p.setInitialFailed(resolvedPeers, bindAddr)
 	}
 
+	// TODO: Don't just dereference.
```

Comment:
We're handling all this differently in the node exporter tls stuff, which you should probably vendor/hack in here to keep things in line

Reply:
For sure. I will align this once https://github.com/prometheus/node_exporter/pull/1277 is merged.